services/xcode-select-service.ts

```diff
@@ -0,0 +1,23 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import {join} from "path";
+
+export class XcodeSelectService implements IXcodeSelectService {
+	constructor(private $childProcess: IChildProcess) {
+	}
+
+	public getDeveloperDirectoryPath(): IFuture<string> {
+		return (() => {
+			let childProcess = this.$childProcess.spawnFromEvent("xcode-select", ["-print-path"], "close").wait();
+			return childProcess.stdout.trim();
+		}).future<string>()();
+	}
+
+	public getContentsDirectoryPath(): IFuture<string> {
+		return (() => {
+			return join(this.getDeveloperDirectoryPath().wait(), "..");
```

Comment:
I prefer using path.join instead of importing only the join. I do not insist on changing this, but the code is more clear for me with path.join